### PR TITLE
CRIMAPP-400,417,567 - List property assets

### PR DESCRIPTION
**Pull request by ripan**
## Description of change
List all the property assets
- Residential
- Commercial
- Land

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-400
https://dsdmoj.atlassian.net/browse/CRIMAPP-417
https://dsdmoj.atlassian.net/browse/CRIMAPP-567

### After changes:
Screenshot 2024-03-25 at 17 24 53

## How to manually test the feature
http://HOST/applications/:id

### CODE REVIEW

File: app/views/casework/crime_applications/sections/_properties.html.erb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+          </dd>
+        </div>
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">
+            <%= label_text(:is_home_address, scope: 'properties', asset: asset) %>
+          </dt>
+          <dd class="govuk-summary-list__value">
+            <%= t(property.is_home_address, scope: 'values') %>
+          </dd>
+        </div>
```
Comment (hiboabd) [inline app/views/casework/crime_applications/sections/_properties.html.erb:95]: I believe these are partner fields. These fields should be hidden if it wasn't asked in apply

File: app/views/casework/crime_applications/sections/_properties.html.erb [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">
+            <%= label_text(:address, scope: 'properties', asset: asset) %>
+          </dt>
+          <dd class="govuk-summary-list__value">
+            <%= render 'address', address: property.address %>
+          </dd>
+        </div>
```
Comment (hiboabd) [inline app/views/casework/crime_applications/sections/_properties.html.erb:103]: I'm getting an error viewing the address if I selected yes to the address being the same as the client's address. I think this needs to display the client's address if there is no property address

File: config/locales/en/casework.yml [hunk 3]
```diff
@@ -168,7 +168,24 @@ en:
...
+      outstanding_mortgage: How much is left to pay on the mortgage?
+      percentage_applicant_owned: What percentage of the %{asset} does your client own?
+      percentage_partner_owned: What percentage of the %{asset} does your client’s partner own?
+      is_home_address: Is the address of the %{asset} the same as your client’s home address?
+      has_other_owners: Does anyone else own part of the %{asset}?
+      address: What is the address of your client’s %{asset}?
+    property_owners:
+      name: What is the name of the %{index} other owner?
+      relationship: What is their relationship to your client?
+      percentage_owned: What percentage of the %{asset} do they own?
```
Comment (hiboabd) [inline config/locales/en/casework.yml:188]: The review designs have different headings for the rows to the apply screens: https://www.figma.com/file/thkvnDBbQbHlSqzt1TE3lk/Crime-Apply?type=design&node-id=8359-48221&mode=design&t=k8CcQlVoOSw0ll4P-0

File: db/schema.rb [hunk 4]
```diff
@@ -20,6 +20,41 @@
   # Note that some types may not work with other database engines. Be careful if changing database.
   create_enum "user_role", ["caseworker", "supervisor", "data_analyst"]
 
+  create_table "crime_applications", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
```
Comment (hiboabd) [inline db/schema.rb:23]: Accidentally pushed?

File: app/views/casework/crime_applications/sections/_properties.html.erb [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+            <dd class="govuk-summary-list__value">
+              <%= property_owner.name %>
+            </dd>
+          </div>
+          <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">
+              <%= label_text(:relationship, scope: 'property_owners') %>
+            </dt>
+            <dd class="govuk-summary-list__value">
+              <%= property_owner.relationship == PropertiesPresenter::OTHER ? property_owner.other_relationship : t(property_owner.relationship, scope: 'values.relationship_type') %>
```
Comment (hiboabd) [inline app/views/casework/crime_applications/sections/_properties.html.erb:126]: I think these values need translating correctly Screenshot 2024-03-21 at 13 21 31

File: config/locales/en/casework.yml [hunk 6]
```diff
@@ -295,7 +312,31 @@ en:
...
+      detached: Detached
+      flat_or_maisonette: Flat or maisonette
+      semidetached: Semi-detached
+      terraced: Terraced
+      other: The type of property is not listed
+    property_type:
+      residential: Residential property
+      commercial: Commercial property
+      land: Land
+    relationship_type:
```
Comment (hiboabd) [inline config/locales/en/casework.yml:329]: The translation for other is missing
Screenshot 2024-03-21 at 13 21 31

File: app/models/crime_application.rb [hunk 7]
```diff
@@ -135,6 +135,10 @@ def outgoings_details
     @outgoings_details ||= OutgoingsDetailsPresenter.present(self[:means_details].outgoings_details)
   end
 
+  def properties
+    @properties ||= PropertiesPresenter.present(self[:means_details].capital_details&.properties || [])
```
Comment (hiboabd) [inline app/models/crime_application.rb:139]: I think it's an empty array by default so you don't need the `|| []`
If a user selects no to say their client doesn't have any assets, what is meant to be displayed in review? I don't think there is a design in the review section

Comment (hiboabd) [reply app/models/crime_application.rb:139]: Actually i found this example for national savings certificate: https://www.figma.com/file/thkvnDBbQbHlSqzt1TE3lk/Crime-Apply?type=design&node-id=11311-44907&mode=design&t=gFQeEeNApBZHvUQ7-0

Screenshot 2024-03-21 at 13 38 01

File: app/views/casework/crime_applications/sections/_properties.html.erb (hunk 2 above)
Comment (ripan) [reply app/views/casework/crime_applications/sections/_properties.html.erb:103]: Displaying client's address can be misleading. Instead we should show address only if client filled the address form in property journey

File: config/locales/en/casework.yml (hunk 6 above)
Comment (ripan) [reply config/locales/en/casework.yml:342]: Need to confirm with designers

File: app/models/crime_application.rb (hunk 7 above)
Comment (ripan) [reply app/models/crime_application.rb:143]: empty assets need to confirm with designers

File: app/views/casework/crime_applications/sections/_properties.html.erb [hunk 8]
```diff
@@ -0,0 +1,155 @@
+<% if properties %>
+  <h2 class="govuk-heading-m">
+    <%= label_text(:properties_title) %>
+  </h2>
+
+  <% if properties.any? %>
+    <% properties.each_with_index do |property| %>
+      <% asset = PropertiesPresenter::PROPERTY_TYPE_MAPPING[property.property_type.to_sym][:display_name]%>
+      <dl class="govuk-summary-list govuk-!-margin-bottom-9">
```
Comment (timpeat) [inline app/views/casework/crime_applications/sections/_properties.html.erb:9]: @ripan might be worth checking out my PR. I've added a component for rendering grouped lists of things:

https://github.com/ministryofjustice/laa-review-criminal-legal-aid/pull/567/files#diff-7ea3a2f1a2bb8203307514304b03264f22a77054911e013cd8f645016dd2f379R1-R22

Comment (ripan) [reply app/views/casework/crime_applications/sections/_properties.html.erb]: UI fixed with new components

File: spec/support/matchers/summary_card.rb [hunk 9]
```diff
@@ -1,6 +1,6 @@
 RSpec::Matchers.define :have_summary_row do |question, answer|
   match do |page|
-    key = page.find('dt', text: question)
+    key = page.find('dt', text: question, match: :prefer_exact)
```
Comment (timpeat) [inline spec/support/matchers/summary_card.rb:3]: 👍

**Review by timpeat — APPROVED** [review]
👍

**Pull request merged**